robot_smach_states/src/robot_smach_states/util/designators/checks.py

```diff
@@ -59,12 +86,39 @@ def check_type(designator_or_value, *allowed_types):
     Traceback (most recent call last):
       ...
     TypeError: ...
+
+    >>> c3 = Designator(["a"], resolve_type=[str], name='c3')
+    >>> check_type(c3, [str])
+    >>> check_type(c3, str, [str])
+    >>> check_type(c3, [str], str)
+    >>> c4 = ["a"]
+    >>> check_type(c4, [str])
+    >>> check_type(c4, str, [str])
+    >>> check_type(c4, [str], str)
+    >>> check_type(c4, [int])  # doctest: +IGNORE_EXCEPTION_DETAIL
+    Traceback (most recent call last):
+      ...
+    TypeError: ...
     """
-    if hasattr(designator_or_value, "resolve_type"): #If its a designator: ...
+    for allowed_type in allowed_types:
+        if isinstance(allowed_type, list):
+            assert len(allowed_type) == 1
+
+    if hasattr(designator_or_value, "resolve_type"):  # If its a designator: ...
         check_resolve_type(designator_or_value, *allowed_types)
     else:
-        if not type(designator_or_value) in allowed_types:
-            raise TypeError("{0} is of type {1} but should be {2}".format(designator_or_value, type(designator_or_value), allowed_types))
+        if isinstance(designator_or_value, list):
+            allowed_list_types = [allowed_type[0] for allowed_type in allowed_types if isinstance(allowed_type, list)]
+
+            if type(designator_or_value[0]) not in allowed_list_types:
+                allowed_list_types_string = ", ".join(map("[{}]".format, allowed_list_types))
+                raise TypeError("{0} is of type [{1}] but should be {2}".format(designator_or_value,
+                                                                                type(designator_or_value[0]),
+                                                                                allowed_list_types_string))
+        else:
+            if not type(designator_or_value) in allowed_types:
```

Comment:
Shouldn't allowed_types in this if also be the the allowed_list_types?

Reply:
No, because the value isn't a list here.